pkg/plugins/platform/utils.go

```diff
@@ -39,50 +40,86 @@ func PrepLocalSiteDirs(base string) error {
 	return nil
 }
 
-// SiteList will prepare and render a list of ddev sites running locally.
-func SiteList(containers []docker.APIContainers) error {
-	apps := map[string]map[string]map[string]string{}
-	var appsFound bool
-
-	for pName := range PluginMap {
-		apps[pName] = map[string]map[string]string{}
-	}
+// GetApps returns a list of ddev applictions keyed by platform.
+func GetApps() map[string][]App {
+	apps := make(map[string][]App)
+	for platformType, instance := range PluginMap {
+		labels := map[string]string{
+			"com.ddev.platform":       instance.ContainerPrefix(),
+			"com.ddev.container-type": "web",
+		}
+		sites, err := util.FindContainersByLabels(labels)
 
-	for _, container := range containers {
-		for _, containerName := range container.Names {
-			for pName, plugin := range PluginMap {
-				if strings.HasPrefix(containerName[1:], plugin.ContainerPrefix()) {
-					util.ProcessContainer(apps[pName], pName, containerName[1:], container)
+		if err == nil {
```

Comment:
Needs handling of err here, and I'd rather see explicit handling of err *first* and then the non-error case. Too hard to follow down otherwise.